prime-router/src/main/kotlin/azure/CheckFunction.kt

```diff
@@ -354,28 +363,77 @@ class CheckFunction : Logging {
             runBlocking {
                 launch {
                     val httpHeaders = theRESTTransport.getHeaders(restTransportType, reportId)
-                    val accessToken: String =
-                        theRESTTransport.getOAuthToken(
+                    var accessToken: String? = null
+
+                    // Check authentication
+                    if (restTransportType.authType == "apiKey") {
+                        val apiKeyCredential = credential as UserApiKeyCredential
+                        httpHeaders["System_ID"] = apiKeyCredential.user
+                        httpHeaders["Key"] = apiKeyCredential.apiKey
+                        accessToken = apiKeyCredential.apiKey
+                    }
+
+                    if (restTransportType.authType == "two-legged" || restTransportType.authType == null) {
+                        // parse headers for any dynamic values, OK needs the report ID
+                        accessToken = theRESTTransport.getOAuthToken(
```

Comment:
This seems to be duplicate logic in the RestTransport. Can we turn this into a function a reuse it in both places, to prevent bugs like this from happening again?